via/services/checkmate.py

```diff
@@ -5,13 +5,20 @@
 from via.exceptions import BadURL
 
 
-class ViaCheckmateClient(CheckmateClient):
-    def __init__(self, request):
-        super().__init__(
-            host=request.registry.settings["checkmate_url"],
-            api_key=request.registry.settings["checkmate_api_key"],
+class CheckmateService:
+    def __init__(self, checkmate_client, allow_all, blocked_for, ignore_reasons):
+        self._checkmate_client = checkmate_client
+        self._allow_all = allow_all
+        self._blocked_for = blocked_for
+        self._ignore_reasons = ignore_reasons
```

Comment:
By using the service pattern the construction of the `CheckmateClient` and `CheckmateService` objects and reading the settings and params that these objects need from the `request` is all moved into the `factory()` function (and its unit tests) which allows the `CheckmateService` class itself to be much simpler: it no longer has access to the `request` and no longer reads `request.registry.settings` and `request.params`, and it's much easier to see the four specific dependencies that it actually has (`checkmate_client`, `allow_all`, `blocked_for` and `ignore_reasons`).